peps/pep-0725.rst

```diff
@@ -445,6 +445,38 @@ Specification
 If metadata is improperly specified then tools MUST raise an error to notify
 the user about their mistake.
 
+DepURL
+------
+
+A DepURL implements a scheme for identifying packages that is meant to be
+portable across packaging ecosystems. Its design is::
+
+    dep:type/namespace/name@version?qualifiers#subpath
```

Comment:
Perhaps:

```suggestion
    dep:{type}/{namespace}/{name}@{version}?{qualifiers}#{subpath}
```

to make it clear what's a fixed string and what's a placeholder.

Also I think we should somehow make it clear which separators are part of which components, but I don't really have a good idea how to express that. So we'd e.g. clarify that `{type}/{name}` is fine whereas `{type}//{name}` is not. Also whether "lone" `@`, `?` and `#` are fine or not.

Reply:
It's quite hard to do as a regex or formal grammar. I do have the sense that someone is going to ask for that though. So I'd rather wait for that to figure out what makes the powers-that-be happy. With the many examples given, it's hard to get it wrong in practice so the main improvement would be to make it formally unambiguous, which f-string type syntax doesn't quite do. Is that okay?